### PR TITLE
feat: language server setup

### PR DESCRIPTION
**Pull request by baszalmstra**
This is the initial setup to be able to community with the Mun language server (see [this PR](https://github.com/mun-lang/mun/pull/210) on the main repo). It currently does nothing more than connecting to the Mun language server if a Mun executable was provided.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import * as vscode from 'vscode';
+
+/**
+ * The `Config` class is a class that holds and maintain the configuration for the ete
```
Comment (Wodann) [inline src/config.ts]: `ete`? (end of the line)

File: src/main.ts [hunk 2]
```diff
@@ -1,5 +1,71 @@
...
+ * @param context The vscode context for this extension
+ */
+export async function activate(context: ExtensionContext) {
+
+    const config = new Config(context);
+
+    // Bootstrap the extension, this returns an optional Mun executable path which we can use to 
+    // initialize a language server with
+    const executablePath = await bootstrap(config);
+    if (executablePath != null) {
```
Comment (Wodann) [inline src/main.ts:21]: Maybe log or throw a pop-up error to signal failure?

**Review by Wodann — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 2 above)
Comment (baszalmstra) [reply src/main.ts:21]: This is not a failure case. I don't want to advertise this too much yet. I was planning on adding a warning when we have something useful.

File: src/config.ts (hunk 1 above)
Comment (baszalmstra) [reply src/config.ts]: Fixed

File: src/main.ts (hunk 2 above)
Comment (Wodann) [reply src/main.ts:21]: Then maybe at least add a `TODO` comment to prevent it from being forgotten

Comment (baszalmstra) [reply src/main.ts:21]: Done

**Review by Wodann — APPROVED** [review]

**Pull request merged**